src/mapblock_mesh.cpp

```diff
@@ -194,37 +195,74 @@ u16 getFaceLight(MapNode n, MapNode n2, v3s16 face_dir, INodeDefManager *ndef)
 	Calculate smooth lighting at the XYZ- corner of p.
 	Both light banks
 */
-static u16 getSmoothLightCombined(const v3s16 &p, MeshMakeData *data)
+static u16 getSmoothLightCombined(const v3s16 &p,
+	const std::array<v3s16,8> &dirs, MeshMakeData *data, bool node_solid)
```

Comment:
wouldn't an `std::vector` or even `const v3s16* dirs` be better here?

Reply:
I don't think so, because the size of 8 is fixed.

Reply:
okay then

Reply:
Just noticed that the size of 8 is *not* fixed: it uses only first 5 entries if `node_solid` is true.

Reply:
@numberZero But to make the ambient occlusion consistent, all 8 would have to be evaluated anyway, wouldn't they?

Reply:
Probably. Unsure yet.